mmcv/ops/diff_iou_rotated.py

```diff
@@ -0,0 +1,296 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+# Adapted from https://github.com/lilanxiao/Rotated_IoU/blob/master/box_intersection_2d.py  # noqa
+# Adapted from https://github.com/lilanxiao/Rotated_IoU/blob/master/oriented_iou_loss.py  # noqa
+import torch
+from torch.autograd import Function
+
+from ..utils import ext_loader
+
+EPSILON = 1e-8
+ext_module = ext_loader.load_ext('_ext',
+                                 ['diff_iou_rotated_sort_vertices_forward'])
+
+
+class SortVertices(Function):
+
+    @staticmethod
+    def forward(ctx, vertices, mask, num_valid):
+        idx = ext_module.diff_iou_rotated_sort_vertices_forward(
+            vertices, mask, num_valid)
+        ctx.mark_non_differentiable(idx)
+        return idx
+
+    @staticmethod
+    def backward(ctx, gradout):
+        return ()
+
+
+def box_intersection_th(corners1: torch.Tensor, corners2: torch.Tensor):
+    """Find intersection points of rectangles.
+    Convention: if two edges are collinear, there is no intersection point
+
+    Args:
+        corners1 (torch.Tensor): B, N, 4, 2
+        corners2 (torch.Tensor): B, N, 4, 2
+
+    Returns:
+        intersections (torch.Tensor): B, N, 4, 4, 2
+        mask (torch.Tensor) : B, N, 4, 4; bool
+    """
+    # build edges from corners
```

Comment:
Add some assert about the input shape would be better.

Reply:
Do we really need it? It is a private function. Only `diff_iou_rotated_2d` and `diff_iou_rotated_3d` are imported from this file.

Reply:
Sure, that's OK.